src/probe/probe.c

```diff
@@ -1021,10 +1066,12 @@ int probe_point_add(uint32_t count, struct probe_point *probe)
 
 				return -EBUSY;
 			}
-		} else if (probe[i].purpose == PROBE_PURPOSE_EXTRACTION) {
+		} else if (probe[i].purpose == PROBE_PURPOSE_EXTRACTION ||
+			   probe[i].purpose == PROBE_PURPOSE_LOGGING) {
 			for (j = 0; j < CONFIG_PROBE_POINTS_MAX; j++) {
 				if (_probe->probe_points[j].stream_tag != PROBE_DMA_INVALID &&
-				    _probe->probe_points[j].purpose == PROBE_PURPOSE_EXTRACTION)
+				    (_probe->probe_points[j].purpose == PROBE_PURPOSE_EXTRACTION ||
+				     _probe->probe_points[j].purpose == PROBE_PURPOSE_LOGGING))
```

Comment:
```c
if (_probe->probe_points[j].stream_tag != PROBE_DMA_INVALID &&
    _probe->probe_points[j].purpose != PROBE_PURPOSE_INJECTION)
```
perhaps? Or can the purpose of the point be other than the valid three at any given time?

Reply:
I added an explicit check with a helper to make the code more readable.